app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseService.kt

```diff
@@ -309,6 +311,22 @@ class FirestoreDatabaseService internal constructor(
         return documentRef.get().await().toObject(UserSchema::class.java)!!.toPublic()
     }
 
+    //TODO Optimize transaction to db
+    override suspend fun setAdminAccess(firebaseUser: FirebaseUser ,adminAccess : Boolean): User {
```

Comment:
Using the [update method](https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentReference#update) would improve this

Reply:
You mean by changing the signature of update method to be the same as setAdmin ? I thought about that but for further implementations I think there's going to be a fundamental difference between setAdmin and update as more checks will be happening to approve of user as admin

Reply:
No, I mean changing how the document is updated by using `documentRef.update(...` instead of `documentRef.set(...`